### ISSUE TITLE
Problem with RequestPathRule and ignore 

### ISSUE THREAD

**Issue by ddurieux** [JimTools/jwt-auth#15]
**Describe the bug**
The ignore list into class RequestPathRule not works.

I do a little debug, and in the __invoke function, the `$this->ignore` is always an empty array, not get my values.

**To Reproduce**

1. use `new RequestPathRule(['/'], ['/FusionResolveIT/api/v1/fusioninventory'])]` but it not ignore my path
2. I have error with message `Token not found. `

I'm on PHP 8.2.26.

**Expected behavior**
I must not have error Token not found for URL '/FusionResolveIT/api/v1/fusioninventory'

**Screenshots**
![image](https://github.com/user-attachments/assets/24f6de6a-3212-419a-9ce4-b23670046f43)

**Package Versions**
2.2.0

**Additional context**
No, sorry

**Comment by JimTools** [comment]
Hi @ddurieux, thanks for raising this ill check this out tomorrow.

**Comment by JimTools** [comment]
@ddurieux I've done some testing with the package and created a very basic slim app. with two routes,: `/FusionResolveIT/api/v1/protected`, `/FusionResolveIT/api/v1/fusioninventory`. this works as expected.

I have two questions for you.

1. Can you provide a curl request for your local development endpoint?
2. I've noticed the path starts with `/FusionResolveIT`, Can you give me some context on how your app is setup and structured. (ie do you're project start in `.` or `/FusionResolveIT`)

**Comment by NeftaliAcosta** [comment]
Hello, I have the same problem, I get this message. Could you help me please?

![Image](https://github.com/user-attachments/assets/94d9875d-cdb8-46f0-be5b-0ee53abcccdc)

This is my default middleware file

```
<?php

use App\Application\Actions\ActionPayload;
use App\Application\Middleware\SessionMiddleware;
use JimTools\JwtAuth\Decoder\FirebaseDecoder;
use JimTools\JwtAuth\Handlers\AfterHandlerInterface;
use JimTools\JwtAuth\Middleware\JwtAuthentication;
use JimTools\JwtAuth\Options;
use JimTools\JwtAuth\Secret;
use Psr\Http\Message\ResponseInterface as Response;
use Psr\Http\Message\ServerRequestInterface as Request;
use Slim\App;

return function (App $app) {
    $key = $_ENV['JWT_AUTH_KEY'];
    $secure = (bool)$_ENV['JWT_SECURE'];

    $afterHandler = new class implements AfterHandlerInterface {
        public function __invoke(Response $response, array $arguments): Response
        {

            return $response;
        }
    };

    $options = new Options(
        isSecure: $secure,
        relaxed: ['', '/', '/auth/login'],
        header: 'Authorization',
        attribute: 'jwt',
        before: null,
        after: $afterHandler
    );

    $decoder = new FirebaseDecoder(
        new Secret($key, 'HS256')
    );

    $middleware = new JwtAuthentication($options, $decoder);

    $app->add(SessionMiddleware::class);

    $app->addMiddleware($middleware);

    $app->add(function (Request $request, $handler) {
        try {
            return $handler->handle($request);
        } catch (Exception $e) {
            $response = new \Slim\Psr7\Response();
            $payload = new ActionPayload(
                401,
                [],
                errorMessage: ["Exception" => $e->getMessage()]
            );
            $response->getBody()->write(json_encode($payload));

            return $response->withHeader("Content-Type", "application/json")->withStatus(401);
        }
    });
};

```


Slim framewok skeleton 4.
php 8.3
OS. Kubuntu 22.04
Package Versions 2.2.0

Thanks

**Comment by JimTools** [comment]
hi @NeftaliAcosta and thanks for providing a snippet, I'll continue to look into this.

I have a quick question based on the screenshot, Are you building a stateful app or a stateless API?

**Comment by ddurieux** [comment]
Hi, I haven't yet check with the information you need, sorry for the delay, I will give you info in the next 4 days.

**Comment by JimTools** [comment]
@ddurieux I have a suspicion that the documentation isn't clear and it's been incorrectly implemented. So I've created a reference  implementation which can be found here [jwt-auth-example](https://github.com/JimTools/jwt-auth-example)

@NeftaliAcosta I can see some issues, the [relaxed](https://jimtools.github.io/jwt-auth/options.html#relaxed) argument in Options is incorrectly being used. this is mostly for development. It will relax the HTTPS check for domains it matches against when `isSecure` is true 

What I think you want to do is provide rules to `JwtAuthentication`

```php
$middleware = new JwtAuthentication(
    new Options(attribute: 'jwt'),
    new FirebaseDecoder(new Secret($key, 'HS256')),
    [new RequestMethodRule(), new RequestPathRule(ignore: ['/auth/login'])],
);
```

**Comment by JimTools** [comment]
I've taken a look at the v2 docs and it doesn't read well. I've created a new issue (#17) to improve the grammar.

**Comment by ddurieux** [comment]
I have restarted from scratch and works nicely, sorry to have taken your precious time :/

**Comment by JimTools** [comment]
@ddurieux don’t worry about it.

I mostly do this for fun, if you have an more issue or if you found an area where the docs 

please feel free to raise an issue PR are also welcome
